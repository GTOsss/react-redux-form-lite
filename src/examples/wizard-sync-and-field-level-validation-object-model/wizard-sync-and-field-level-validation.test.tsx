import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers} from 'redux';
import ReduxThunkTester from 'redux-thunk-tester';
import Field from '../../field';
import WizardSyncAndFieldLevelValidation from './wizard-sync-and-field-level-validation';
import {reducer} from '../../index';
import get = Reflect.get;

interface IValuesForm1 {
  firstName?: string;
  lastName?: string;
}

interface IValuesForm2 {
  phone?: string;
  email?: string;
}

interface IValuesForm3 {
  dob?: string;
  hobby?: string;
}

const renderComponent = (onSubmit?) => {
  const reduxThunkTester = new ReduxThunkTester();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}><WizardSyncAndFieldLevelValidation onSubmit={onSubmit} /></Provider>,
  );

  return {
    reduxThunkTester,
    getStep1: (): IReduxFormState<IValuesForm1> => store.getState().reduxForm.step1 as IReduxFormState<IValuesForm1>,
    getStep2: (): IReduxFormState<IValuesForm2> => store.getState().reduxForm.step2 as IReduxFormState<IValuesForm2>,
    getStep3: (): IReduxFormState<IValuesForm3> => store.getState().reduxForm.step3 as IReduxFormState<IValuesForm3>,
    getWizardExample: (): IReduxFormWizard<IValuesForm1 & IValuesForm2 & IValuesForm3> =>
      store.getState().reduxForm.wizardExample as IReduxFormWizard<IValuesForm1 & IValuesForm2 & IValuesForm3>,
    component,
  };
};

describe('<WizardSyncAndFieldLevelValidationObjectModel />', () => {
  test('Render WizardSyncAndFieldLevelValidation (inputs)', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render WizardSyncAndFieldLevelValidation: store of wizard', () => {
    const {getWizardExample} = renderComponent();
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit failed on page 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit failed on page 1 (store)', () => {
    const {component, getStep1} = renderComponent();
    component.find('form').simulate('submit');
    expect(getStep1()).toMatchSnapshot();
  });

  test('submit failed on page 1, incorrect firstName (store)', () => {
    const {component, getStep1} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'te'}});
    component.find('form').simulate('submit');
    expect(getStep1()).toMatchSnapshot();
  });

  test('submit failed on page 1 (errors in wizard === errors in step1)', () => {
    const {component, getWizardExample, getStep1} = renderComponent();
    component.find('form').simulate('submit');
    expect(getStep1().form.errorsMap).toStrictEqual(getWizardExample().wizard.errorsMap);
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on page 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on page 1 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on page 1 (store of step1)', () => {
    const {component, getStep1} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getStep1()).toMatchSnapshot();
  });

  test('submit success on page 1 (store of step2)', () => {
    const {component, getStep2} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getStep2()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2 (errors in wizard === errors in step2)', () => {
    const {component, getWizardExample, getStep2} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getStep2().form.errorsMap).toStrictEqual(getWizardExample().wizard.errorsMap);
  });

  test('submit success on 1, failed on 2, click prevent page (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent page, submit success on 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent page (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent page, submit success on 1 (store 2)', () => {
    const {component, getStep2} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    component.find('form').simulate('submit');
    expect(getStep2()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('button').at(0).simulate('click');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('button').at(0).simulate('click');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2 (store 2)', () => {
    const {component, getStep2} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    expect(getStep2()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, 2 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, 2 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, failed on 2, click prevent, success on 1 (render inputs)', () => {
    const {component} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('button').at(0).simulate('click');
    component.find('form').simulate('submit');
    expect(component.find('input')).toMatchSnapshot();
  });

  test('submit success on 1, 2, 3 (store of wizard)', () => {
    const {component, getWizardExample} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getWizardExample()).toMatchSnapshot();
  });

  test('submit success on 1, 2, 3 (store 3)', () => {
    const {component, getStep3, reduxThunkTester} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(getStep3()).toMatchSnapshot();
  });

  test('submit success on page 1, 2, 3 (onSubmit)', () => {
    const onSubmit = jest.fn();
    const {component} = renderComponent(onSubmit);
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    component.find('input').at(0).simulate('change', {target: {value: 'test'}});
    component.find('input').at(1).simulate('change', {target: {value: 'test'}});
    component.find('form').simulate('submit');
    expect(onSubmit.mock.calls).toMatchSnapshot();
  });

  test('submit field-level failed on 1', () => {
    const {component, getWizardExample, getStep1, reduxThunkTester} = renderComponent();
    component.find('input').at(0).simulate('change', {target: {value: 't'}});
    component.find('form').simulate('submit');
    expect(getWizardExample().wizard.errorsMap).toStrictEqual(getStep1().form.errorsMap);
    expect(getWizardExample()).toMatchSnapshot();
  });
});
