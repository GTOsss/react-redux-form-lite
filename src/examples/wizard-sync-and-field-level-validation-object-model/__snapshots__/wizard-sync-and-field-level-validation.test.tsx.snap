// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> Render WizardSyncAndFieldLevelValidation (inputs) 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> Render WizardSyncAndFieldLevelValidation: store of wizard 1`] = `
Object {
  "values": Object {
    "profile": Object {
      "firstName": undefined,
      "lastName": undefined,
    },
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit failed on page 1 (errors in wizard === errors in step1) 1`] = `
Object {
  "values": Object {
    "profile": Object {
      "firstName": undefined,
      "lastName": undefined,
    },
  },
  "wizard": Object {
    "errorsMap": Object {
      "profile": Object {
        "firstName": "Field required.",
        "lastName": "Field required.",
      },
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit failed on page 1 (render inputs) 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit failed on page 1 (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {
      "profile": Object {
        "firstName": "Field required.",
        "lastName": "Field required.",
      },
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "Field required.",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "Field required.",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "profile": Object {
      "firstName": undefined,
      "lastName": undefined,
    },
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit failed on page 1, incorrect firstName (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {
      "profile": Object {
        "firstName": "The minimum length of the value must be 3.",
        "lastName": "Field required.",
      },
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "The minimum length of the value must be 3.",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "Field required.",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "profile": Object {
      "firstName": "te",
      "lastName": undefined,
    },
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit field-level failed on 1 1`] = `
Object {
  "values": Object {
    "profile": Object {
      "firstName": "t",
      "lastName": undefined,
    },
  },
  "wizard": Object {
    "errorsMap": Object {
      "profile": Object {
        "firstName": "The minimum length of the value must be 3.",
        "lastName": "Field required.",
      },
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, 2 (render inputs) 1`] = `
Array [
  <input
    name="additionInfo.dob"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Date of birthday"
    type="text"
    value=""
  />,
  <input
    name="additionInfo.hobby"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Hobby"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, 2 (store of wizard) 1`] = `
Object {
  "values": Object {
    "additionInfo": Object {
      "dob": undefined,
      "hobby": undefined,
    },
    "contacts": Object {
      "phone": "test",
    },
    "profile": Object {
      "email": "test",
      "firstName": "test",
      "lastName": "test",
    },
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, 2, 3 (store 3) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "additionInfo": Object {
      "dob": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "hobby": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "additionInfo": Object {
      "dob": "test",
      "hobby": "test",
    },
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, 2, 3 (store of wizard) 1`] = `
Object {
  "values": Object {
    "additionInfo": Object {
      "dob": "test",
      "hobby": "test",
    },
    "contacts": Object {
      "phone": "test",
    },
    "profile": Object {
      "email": "test",
      "firstName": "test",
      "lastName": "test",
    },
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2 (render inputs) 1`] = `
Array [
  <input
    name="contacts.phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value=""
  />,
  <input
    name="profile.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2 (store 2) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {
      "contacts": Object {
        "phone": "Field required.",
      },
      "profile": Object {
        "email": "Field required.",
      },
    },
    "focused": false,
    "hasErrors": true,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "phone": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "Field required.",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "Field required.",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "phone": undefined,
    },
    "profile": Object {
      "email": undefined,
    },
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2 (store of wizard) 1`] = `
Object {
  "values": Object {
    "contacts": Object {
      "phone": undefined,
    },
    "profile": Object {
      "email": undefined,
      "firstName": "test",
      "lastName": "test",
    },
  },
  "wizard": Object {
    "errorsMap": Object {
      "contacts": Object {
        "phone": "Field required.",
      },
      "profile": Object {
        "email": "Field required.",
      },
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2, click prevent (render inputs) 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value="test"
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value="test"
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2, click prevent (store of wizard) 1`] = `
Object {
  "values": Object {
    "contacts": Object {
      "phone": "test",
    },
    "profile": Object {
      "email": undefined,
      "firstName": "test",
      "lastName": "test",
    },
  },
  "wizard": Object {
    "errorsMap": Object {
      "profile": Object {
        "email": "Field required.",
      },
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2, click prevent page (render inputs) 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value="test"
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value="test"
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2, click prevent page (store of wizard) 1`] = `
Object {
  "values": Object {
    "contacts": Object {
      "phone": undefined,
    },
    "profile": Object {
      "email": undefined,
      "firstName": "test",
      "lastName": "test",
    },
  },
  "wizard": Object {
    "errorsMap": Object {
      "contacts": Object {
        "phone": "Field required.",
      },
      "profile": Object {
        "email": "Field required.",
      },
    },
    "hasErrors": true,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2, click prevent page, submit success on 1 (render inputs) 1`] = `
Array [
  <input
    name="contacts.phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value=""
  />,
  <input
    name="profile.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2, click prevent page, submit success on 1 (store 2) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "phone": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "phone": undefined,
    },
    "profile": Object {
      "email": undefined,
    },
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on 1, failed on 2, click prevent, success on 1 (render inputs) 1`] = `
Array [
  <input
    name="contacts.phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value="test"
  />,
  <input
    name="profile.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on page 1 (render inputs) 1`] = `
Array [
  <input
    name="contacts.phone"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Phone"
    type="text"
    value=""
  />,
  <input
    name="profile.email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />,
]
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on page 1 (store of step1) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": true,
    "warningsMap": Object {},
  },
  "meta": Object {
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": true,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "profile": Object {
      "firstName": "test",
      "lastName": "test",
    },
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on page 1 (store of step2) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "contacts": Object {
      "phone": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
    "profile": Object {
      "email": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "contacts": Object {
      "phone": undefined,
    },
    "profile": Object {
      "email": undefined,
    },
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on page 1 (store of wizard) 1`] = `
Object {
  "values": Object {
    "contacts": Object {
      "phone": undefined,
    },
    "profile": Object {
      "email": undefined,
      "firstName": "test",
      "lastName": "test",
    },
  },
  "wizard": Object {
    "errorsMap": Object {},
    "hasErrors": false,
    "hasWarnings": false,
    "warningsMap": Object {},
  },
}
`;

exports[`<WizardSyncAndFieldLevelValidationObjectModel /> submit success on page 1, 2, 3 (onSubmit) 1`] = `
Array [
  Array [
    Object {
      "actions": Object {
        "arrayPush": [Function],
        "arrayRemove": [Function],
        "blur": [Function],
        "change": [Function],
        "changeSubmitted": [Function],
        "focus": [Function],
        "registerField": [Function],
        "registerForm": [Function],
        "removeField": [Function],
        "removeForm": [Function],
        "resetForm": [Function],
        "setInitialValues": [Function],
        "updateFormState": [Function],
        "updateValidateAndWarningMessages": [Function],
        "updateValidateMessage": [Function],
        "updateValidateMessages": [Function],
        "updateWarningMessage": [Function],
        "updateWarningMessages": [Function],
      },
      "state": Object {
        "form": Object {
          "activeField": "",
          "blurred": false,
          "changed": true,
          "errorsMap": Object {},
          "focused": false,
          "hasErrors": false,
          "hasWarnings": false,
          "submitted": true,
          "warningsMap": Object {},
        },
        "meta": Object {
          "additionInfo": Object {
            "dob": Object {
              "active": false,
              "blurred": false,
              "changed": true,
              "error": "",
              "focused": false,
              "warning": "",
            },
            "hobby": Object {
              "active": false,
              "blurred": false,
              "changed": true,
              "error": "",
              "focused": false,
              "warning": "",
            },
          },
        },
        "values": Object {
          "additionInfo": Object {
            "dob": "test",
            "hobby": "test",
          },
        },
      },
      "values": Object {
        "additionInfo": Object {
          "dob": "test",
          "hobby": "test",
        },
      },
      "wizard": Object {
        "values": Object {
          "additionInfo": Object {
            "dob": "test",
            "hobby": "test",
          },
          "contacts": Object {
            "phone": "test",
          },
          "profile": Object {
            "email": "test",
            "firstName": "test",
            "lastName": "test",
          },
        },
        "wizard": Object {
          "errorsMap": Object {},
          "hasErrors": false,
          "hasWarnings": false,
          "warningsMap": Object {},
        },
      },
    },
  ],
]
`;
